### DISCUSSION TITLE
Is there a function in HIGHS to convert HIGHS solution file format to GLPK solution file format using highspy?

### DISCUSSION THREAD

**Discussion by hbshrestha** [ERGO-Code/HiGHS#2072] [Q&A] [answered]
Hello. I am currently using HIGHS to solve a mps file and get a solution file using the following code:

```
import highspy

h = highspy.Highs()
filename = 'abc.mps'
h.readModel(filename)

h.setOptionValue("solver","ipm")

h.run()

h.writeSolution("abc_highs_solution.sol", 1)
```

The solution file format of HIGHS looks as shown below. It has two different sections for columns and rows:
![image](https://github.com/user-attachments/assets/a387a22d-8ecf-47b9-b55d-0d82d6fc1b10)

I am putting some sample rows and columns below:

Columns
    Index Status        Lower        Upper       Primal         Dual  Name
        0     BS            0        0.012         0.01            0  XD001.....aaa023
        1     BS            0        0.012        0.012            0  XD001.....aaa024
        2     BS            0        0.012        0.012            0  XD001.....aaa025
        3     BS            0        0.012        0.012            0  XD001.....aaa026
        4     BS            0        0.012        0.012            0  XD001.....aaa027
        5     BS            0        0.012        0.012            0  XD001.....aaa028
Rows
    Index Status        Lower        Upper       Primal         Dual  Name
        0     BS         -inf      75684.3            0           -0  Is.Ba........026
        1     BS         -inf      75684.3            0           -0  Is.Ba........027
        2     BS         -inf      75684.3            0           -0  Is.Ba........028
        3     BS         -inf      75684.3            0           -0  Is.Ba........029
        4     BS         -inf      75684.3            0           -0  Is.Ba........030
        5     BS         -inf      75684.3            0           -0  Is.Ba........031

I need to get the solution into the GLPK solution file format which is as shown. It also has two sections for rows and columns.
![image](https://github.com/user-attachments/assets/befac0e0-1ee7-4807-ac78-959ce5689041)

I am putting some sample rows and columns below:

PROBLEM NAME     abc.mps              
 DATA    NAME                         
 OBJECTIVE VALUE  3.619197E+07  
 STATUS           OPTIMAL SOLN     
 ITERATION        59011

 OBJECTIVE        func                 (MIN)
 RHS                                  
 RANGES                               
 BOUNDS                               

 SECTION 1 - ROWS

  NUMBER  ......ROW....... AT  ...ACTIVITY...  SLACK ACTIVITY  ..LOWER LIMIT.  ..UPPER LIMIT.  .DUAL ACTIVITY

       1  func             BS    3.619197E+07   -3.619197E+07            NONE            NONE               1
       2  Is.Ba........026 BS               0         75684.3            NONE         75684.3              -0
       3  Is.Ba........027 BS               0         75684.3            NONE         75684.3              -0
       4  Is.Ba........028 BS               0         75684.3            NONE         75684.3              -0
       5  Is.Ba........029 BS               0         75684.3            NONE         75684.3              -0

SECTION 2 - COLUMNS

  NUMBER  .....COLUMN..... AT  ...ACTIVITY...  ..INPUT COST..  ..LOWER LIMIT.  ..UPPER LIMIT.  .REDUCED COST.

  139322  XD001.....aaa023 BS            0.01               0               0           0.012               0
  139323  XD001.....aaa024 BS            0.01               0               0           0.012               0
  139324  XD001.....aaa025 BS            0.01               0               0           0.012               0
  139325  XD001.....aaa026 BS            0.01               0               0           0.012               0
  139326  XD001.....aaa027 BS            0.01               0               0           0.012               0

Is there a functionality available to convert the HIGHS solution file to the GLPK solution file format?

**Comment by jajhall** [comment, marked as answer]
I think so. I wrote code to write out the solution in `Glpsol` format (raw and "pretty"), and this is achieved by passing 2 (`kSolutionStyleGlpsolRaw`) or 3 (`kSolutionStyleGlpsolPretty`) as the style argument to `writeSolution(filename, style)`

**Reply by hbshrestha** [reply]
Thank you for the response. I will check it out.

**Reply by hbshrestha** [reply]
By passing 3, I am getting very similar to the GLPK format I shared in the question. Header names are different.
`h.writeSolution("abc_highs_solution.sol", 3)`

The new solution format looks as shown:
![image](https://github.com/user-attachments/assets/d581bff2-1ea1-4a1b-8c0c-b00428db8e98)
